### PR TITLE
Parse manifest

### PR DESCRIPTION
**Pull request by chm-diederichs**

### CODE REVIEW

File: lib/manifest.js [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+  }
+}
+
+function signed (signable, signature, idx) {
+  for (let i = 0; i < idx.length; i++) {
+    const indexer = idx[i]
+    if (!indexer.verify(signable, signature)) continue
+
+    const swap = idx.pop()
+    if (indexer !== swap) idx[i] = swap
```
Comment (mafintosh) [inline lib/manifest.js:194]: idx[i--] = swap

File: index.js [hunk 2]
```diff
@@ -74,10 +75,11 @@ module.exports = class Hypercore extends EventEmitter {
     this.writable = false
     this.opened = false
     this.closed = false
+    this.sign = null
```
Comment (mafintosh) [inline index.js:78]: dont think this is used

File: index.js [hunk 3]
```diff
@@ -250,15 +261,14 @@ module.exports = class Hypercore extends EventEmitter {
...
-
+    if (!this.keyPair) this.keyPair = o.keyPair
     this.crypto = o.crypto
     this.id = o.id
     this.key = o.key
     this.core = o.core
     this.replicator = o.replicator
     this.encryption = o.encryption
-    this.writable = !this._readonly && !!this.auth && !!this.auth.sign
+    this.writable = !this._readonly || !!(this.keyPair && this.keyPair.secretKey)
```
Comment (mafintosh) [inline index.js:271]: should be &&

File: lib/manifest.js (hunk 1 above)
Comment (chm-diederichs) [reply lib/manifest.js:194]: we return immediately after, still good to have it in?

Comment (mafintosh) [reply lib/manifest.js:194]: ah ok! yes, nvm then

**Pull request merged**